src/test/fixtures/fake-user-store.ts

```diff
@@ -131,13 +131,13 @@ class UserStoreMock implements IUserStore {
     upsert(user: ICreateUser): Promise<IUser> {
         this.data.splice(this.data.findIndex((u) => u.email === user.email));
         this.data.push({
+            ...(user as IUser),
             id: this.data.length + 1,
             createdAt: new Date(),
             isAPI: false,
             permissions: [],
             loginAttempts: 0,
             imageUrl: '',
```

Comment:
Oh, this is interesting. Doesn't this change the behavior here? Specifically, the `id` would now get overwritten, whereas previously it wouldn't

Reply:
![image](https://github.com/Unleash/unleash/assets/455064/c51e19b6-7804-4dca-84d5-61f29724a244)
The thing is that switching the order gives errors because many properties are overridden (id should always be defined in the user object so there's no point in defining it)...

I tried removing the as IUser and I get a different kind of error: 
![image](https://github.com/Unleash/unleash/assets/455064/67222314-b4d3-4646-970b-ec20068ca6e7)

but I found a more elegant way of solving both. Commit incoming

Reply:
I believe this fixes your concern: 8b037af924368c13a4c5082b3263e5f386cf977d

Thanks for the challenge :hugs:

Reply:
Nice! But wouldn't that still override `id` if it's present on the user object? And in a way that's different from before, where the user's `id` would be overridden by `this.data.length + 1`?

Reply:
We're probably not sending the id using ts-ignore or ts-expect error. And if we're already sending it then it should be fine because it was sent before. I believe these values act as a default in case some of the values are not provided in the parameter

Reply:
So it's actually same behavior as before